sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -107,7 +107,6 @@ class FileScanRDD(
         val nextElement = currentIterator.next()
         // TODO: we should have a better separation of row based and batch based scan, so that we
         // don't need to run this `if` for every record.
-        val preNumRecordsRead = inputMetrics.recordsRead
```

Comment:
Looks fine but it would have been better which PR removed the last usage of this in order to make any mistake.

Reply:
Changed from [SPARK-25674](https://github.com/apache/spark/commit/4cee191c04f14d7272347e4b29201763c6cfb6bf#diff-b7b097f8cec6a7ae6640f9ecd6d4ac14ed304ad7c8db802a3ec2c0983535e157R113).

This PR directly uses `inputMetrics.recordsRead` instead of using `preNumRecordsRead`